File: C#-OOP/Models/Message.cs
using CSharp_OOP.Services;
using System;

namespace CSharp_OOP.Models
{
    public class Message
    {
		private String message; 

		public Message(String message)
		{ 
			this.message = message;
		}

		public void PrintOut(IPrinter printer)
		{ 
			printer.PrintOut(this.ToString());
		}

		public override String ToString()
		{
			return this.message;
		}
	}
}

File: PHP-OOP/src/models/Message.php
<?php
namespace app\models;

/*
 * class Message 
 * berisi pesan string yang ingin ditampilkan
 */
class Message 
{ 
	private $message; 

	public function __construct($message) { 
		$this->message = $message; 
	} 

	public function printOut($printer) { 
		$printer->printOut($this->toString()); 
	} 

	public function toString() { 
		return $this->message; 
	} 
} 

